modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -242,7 +242,21 @@ const addBrToBlockIfNeeded = (dom, block) => {
   }
 };
 
-const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>) => {
+const findParent = (editor: Editor, container: Node, stepUntilFinding?: string) => {
```

Comment:
Maybe this could take a predicate instead that way this parameter wouldn't need to be optional. Then it would just be  `insertBlock(Fun.always)` and for the blockquote `insertBlock((node) => node.tagName === 'BLOCKQUOTE'` and then you could use PredicateFind.closest(SugarElement.fromDom(container), (node) => editor.dom.isBlock(node.dom) && predicate(node.dom)) and not need to have this `findDirectParent` fallback.

So the entire function becomes something like:
```
const findParent = (editor: Editor, container: Node, predicate: (node: DOMNode) => boolean) => {
  const isRoot = (el: SugarElement<Node>) => el.dom === editor.getBody();
  return PredicateFind.closest(SugarElement.fromDom(container), (node) => editor.dom.isBlock(node.dom) && predicate(node.dom), isRoot).map((element) => element.dom);
};
```

Also we have defined this isRoot so many times it starting to become ridiculous. Maybe we should have a `EditorBody.ts` (I could have sworn that we did have that at some point). That module could have a curried isRoot function so you could do `PredicateFind.closest(.. EditorBody.isRoot(editor))`.

Also if we want to marry these two worlds the sugar world and the dom world at some point I could have sworn I saw someone adding a `SugarElement.toDom` function that just unwraps the node like that. I think it was Josh maybe that was in some PR that never made it. Obviously we shouldn't overuse that but it would be useful here and there especially in mixed code like this.